java/source/flight.rst

```diff
@@ -0,0 +1,1061 @@
+
+.. contents::
+
+Simple VectorSchemaRoot storage service with Arrow Flight
+=========================================================
+
+We'll implement a service to transfer batches flow through VectorSchemaRoot.
+
+Flight Client and Server
+************************
+
+.. code-block:: java
+
+    import org.apache.arrow.flight.Action;
+    import org.apache.arrow.flight.AsyncPutListener;
+    import org.apache.arrow.flight.Criteria;
+    import org.apache.arrow.flight.FlightClient;
+    import org.apache.arrow.flight.FlightDescriptor;
+    import org.apache.arrow.flight.FlightEndpoint;
+    import org.apache.arrow.flight.FlightInfo;
+    import org.apache.arrow.flight.FlightServer;
+    import org.apache.arrow.flight.FlightStream;
+    import org.apache.arrow.flight.Location;
+    import org.apache.arrow.flight.NoOpFlightProducer;
+    import org.apache.arrow.flight.PutResult;
+    import org.apache.arrow.flight.Result;
+    import org.apache.arrow.flight.Ticket;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorLoader;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.VectorUnloader;
+    import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+
+    import java.io.IOException;
+    import java.nio.charset.StandardCharsets;
+    import java.util.ArrayList;
+    import java.util.Arrays;
+    import java.util.Collections;
+    import java.util.HashMap;
+    import java.util.Iterator;
+    import java.util.List;
+    import java.util.Map;
+
+    class DataInMemory {
+        private List<ArrowRecordBatch> listArrowRecordBatch;
+        private Schema schema;
+        private Long rows;
+        public DataInMemory(List<ArrowRecordBatch> listArrowRecordBatch, Schema schema, Long rows) {
+            this.listArrowRecordBatch = listArrowRecordBatch;
+            this.schema = schema;
+            this.rows = rows;
+        }
+        public List<ArrowRecordBatch> getListArrowRecordBatch() {
+            return listArrowRecordBatch;
+        }
+        public Schema getSchema() {
+            return schema;
+        }
+        public Long getRows() {
+            return rows;
+        }
+    }
+
+    // Server
+    Location location = Location.forGrpcInsecure("0.0.0.0", 33333);
+    Map<FlightDescriptor, DataInMemory> dataInMemory = new HashMap<>();
+    Map<String, DataInMemory> mapPojoFlightDataInMemory = new HashMap<>();
+    List<ArrowRecordBatch> listArrowRecordBatch = new ArrayList<>();
+    try (RootAllocator allocator = new RootAllocator(Long.MAX_VALUE)){
+        FlightServer flightServer = FlightServer.builder(allocator, location, new NoOpFlightProducer(){
```

Comment:
Instead of an anonymous class, define a subclass up above and make properties like `dataInMemory` its properties.

Reply:
New CookbookProducer class created